pkg/service/manifest/storage/bolt.go

```diff
@@ -83,10 +84,78 @@ func (b BoltManifestStorage) GetManifests() ([]StoredManifest, error) {
 }
 
 func (b BoltManifestStorage) DeleteManifest(id string) error {
-	if err := b.db.Delete(namespace, id); err != nil {
+	if err := b.db.Delete(manifestNamespace, id); err != nil {
 		errMsg := fmt.Sprintf("could not delete manifest: %s", id)
 		logrus.WithError(err).Error(errMsg)
 		return errors.Wrapf(err, errMsg)
 	}
 	return nil
 }
+
+func (b BoltManifestStorage) StoreApplication(application StoredApplication) error {
+	id := application.Application.Application.ID
+	if id == "" {
+		err := errors.New("could not store application without an ID")
+		logrus.WithError(err).Error()
+		return err
+	}
+	applicationBytes, err := json.Marshal(application)
+	if err != nil {
+		errMsg := fmt.Sprintf("could not store application: %s", id)
+		logrus.WithError(err).Error(errMsg)
+		return errors.Wrapf(err, errMsg)
+	}
+	return b.db.Write(applicationNamespace, id, applicationBytes)
+}
+
+func (b BoltManifestStorage) GetApplication(id string) (*StoredApplication, error) {
+	applicationBytes, err := b.db.Read(applicationNamespace, id)
+	if err != nil {
+		errMsg := fmt.Sprintf("could not get application: %s", id)
+		logrus.WithError(err).Error(errMsg)
+		return nil, errors.Wrapf(err, errMsg)
+	}
+	if len(applicationBytes) == 0 {
+		err := fmt.Errorf("application not found with id: %s", id)
+		logrus.WithError(err).Error("could not get application from storage")
+		return nil, err
+	}
+	var stored StoredApplication
+	if err := json.Unmarshal(applicationBytes, &stored); err != nil {
+		errMsg := fmt.Sprintf("could not unmarshal stored application: %s", id)
+		logrus.WithError(err).Error(errMsg)
+		return nil, errors.Wrapf(err, errMsg)
+	}
+	return &stored, nil
+}
+
+// GetApplications attempts to get all stored applications. It will return those it can even if it has trouble with some.
+func (b BoltManifestStorage) GetApplications() ([]StoredApplication, error) {
+	gotApplications, err := b.db.ReadAll(applicationNamespace)
+	if err != nil {
+		errMsg := "could not get all applications"
+		logrus.WithError(err).Error(errMsg)
+		return nil, errors.Wrap(err, errMsg)
+	}
+	if len(gotApplications) == 0 {
```

Comment:
is this case handled by the caller properly? meaning, no error?

Reply:
currently it functions similarly to GetCredentials. If the len is 0 then it will return null.

Meaning when we do a GetCredentialsBySubject with postman and there are no credentials our response is null

<img width="270" alt="image" src="https://user-images.githubusercontent.com/5314059/190449673-72aa9a8a-e988-4071-8597-64e921a01b3d.png">

Reply:
seems fine to me